src/itest/java/uk/gov/companieshouse/pscdataapi/steps/PscDataSteps.java

```diff
@@ -0,0 +1,101 @@
+package uk.gov.companieshouse.pscdataapi.steps;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.cucumber.java.After;
+import io.cucumber.java.Before;
+import io.cucumber.java.en.Given;
+import io.cucumber.java.en.Then;
+import io.cucumber.java.en.When;
+import org.assertj.core.api.Assertions;
+import org.mockito.MockitoAnnotations;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.web.client.TestRestTemplate;
+import org.springframework.data.mongodb.core.MongoTemplate;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static uk.gov.companieshouse.pscdataapi.config.AbstractMongoConfig.mongoDBContainer;
+
+import uk.gov.companieshouse.pscdataapi.config.CucumberContext;
+import uk.gov.companieshouse.pscdataapi.util.FileReaderUtil;
+import uk.gov.companieshouse.pscdataapi.repository.CompanyPscRepository;
+
+import java.io.IOException;
+import java.util.Collections;
+
+public class PscDataSteps {
+    private String contextId;
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @Autowired
+    private TestRestTemplate restTemplate;
+
+    @Autowired
+    private MongoTemplate mongoTemplate;
+
+    @Autowired
+    private CompanyPscRepository companyPscRepository;
+
+    @Before
+    public void dbCleanUp(){
+        if (!mongoDBContainer.isRunning()) {
+            mongoDBContainer.start();
+        }
+        companyPscRepository.deleteAll();
+        MockitoAnnotations.initMocks(this);
+    }
+
+    @Given("Psc data api service is running")
+    public void theApplicationRunning() {
+        assertThat(restTemplate).isNotNull();
+    }
+
+    @Given("the database is down")
+    public void the_psc_data_db_is_down() {
+        mongoDBContainer.stop();
+    }
+
+    @Then("I should receive {int} status code")
+    public void i_should_receive_status_code(Integer statusCode) {
+        int expectedStatusCode = CucumberContext.CONTEXT.get("statusCode");
+        Assertions.assertThat(expectedStatusCode).isEqualTo(statusCode);
+    }
+
+    @When("I send a PUT request with payload {string} file for company number {string} with notification id {string}")
```

Comment:
I wouldn't bother passing the company number through, set up the company number as a constant at the top of the steps file, there's no reason for it to vary between tests.